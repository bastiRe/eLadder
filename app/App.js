import React from "react";
import { Platform, StatusBar, StyleSheet, View } from "react-native";
import { ApolloProvider } from "react-apollo";
import { Amplitude, AppLoading, Font, Constants } from "expo";
import { Ionicons, Feather, MaterialIcons } from "@expo/vector-icons";
import Sentry from "sentry-expo";

import { client, persistor } from "./createApolloClient";
import RootNavigation from "./navigation/RootNavigation";

const { sentryUrl, amplitudeKey } = Constants.manifest.extra;

Sentry.config(sentryUrl).install();
Amplitude.initialize(amplitudeKey);

class App extends React.Component {
  state = {
    isLoadingComplete: false
  };

  async _prepareApp() {
    await Promise.all([this._loadCache(), ...this._loadFonts()]);
  }

  _loadFonts() {
    const fonts = [Ionicons.font, Feather.font, MaterialIcons.font];
    return fonts.map(font => Font.loadAsync(font));
  }

  async _loadCache() {
    await persistor.restore();
  }

  render() {
    if (!this.state.isLoadingComplete) {
      return (
        <AppLoading
          startAsync={this._loadCache}
          onFinish={() => this.setState({ isLoadingComplete: true })}
        />
      );
    }

    Amplitude.logEvent("OpenApp");

    return (
      <ApolloProvider client={client}>
        <View style={styles.container}>
          {Platform.OS === "ios" && <StatusBar barStyle="light-content" />}
          <RootNavigation />
        </View>
      </ApolloProvider>
    );
  }
}

export default App;

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: "#fff"
  }
});
