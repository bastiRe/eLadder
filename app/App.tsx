import React, { useState, useEffect, useCallback } from "react";
import { QueryClient, QueryClientProvider } from "react-query";
import { LeagueIdsProvider } from "./context/LeagueIds";
import { StatusBar, Platform, View } from "react-native";
import * as SplashScreen from "expo-splash-screen";
import Constants from "expo-constants";
import * as Amplitude from "expo-analytics-amplitude";
import * as Sentry from "sentry-expo";

import { AppNavigator } from "./navigation/AppNavigator";

const { sentryDsn, amplitudeKey } = Constants.manifest.extra;

Sentry.init({
  dsn: sentryDsn,
  enableInExpoDevelopment: true,
  debug: true // If `true`, Sentry will try to print out useful debugging information if something goes wrong with sending the event. Set it to `false` in production
});

Amplitude.initializeAsync(amplitudeKey);

const queryClient = new QueryClient();

function App() {
  const [appIsReady, setAppIsReady] = useState(false);

  useEffect(() => {
    Amplitude.logEventAsync("OpenApp");
    async function prepare() {
      try {
        // Keep the splash screen visible while we fetch resources
        await SplashScreen.preventAutoHideAsync();
      } catch (e) {
        console.warn(e);
      } finally {
        // Tell the application to render
        setAppIsReady(true);
      }
    }

    prepare();
  }, []);

  const onLayoutRootView = useCallback(async () => {
    if (appIsReady) {
      await SplashScreen.hideAsync();
    }
  }, [appIsReady]);

  if (!appIsReady) {
    return null;
  }

  return (
    <LeagueIdsProvider>
      <QueryClientProvider client={queryClient}>
        {Platform.OS === "ios" && <StatusBar barStyle="light-content" />}
        <View onLayout={onLayoutRootView} style={{ flex: 1 }}>
          <AppNavigator />
        </View>
      </QueryClientProvider>
    </LeagueIdsProvider>
  );
}

export default App;
