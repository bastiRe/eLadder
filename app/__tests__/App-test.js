import 'react-native';
import renderer from 'react-test-renderer';
import React from 'react';
import App from '../App';

it('renders the loading screen', async () => {
  const tree = renderer.create(<App />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders the root without loading screen', async () => {
  const tree = renderer.create(<App skipLoadingScreen />).toJSON();
  expect(tree).toMatchSnapshot();
});
