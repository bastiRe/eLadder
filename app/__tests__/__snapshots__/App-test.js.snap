// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the loading screen 1`] = `<ExponentAppLoading />`;

exports[`renders the root without loading screen 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
    }
  }
>
  <View
    onLayout={[Function]}
    style={
      Array [
        Object {
          "flex": 1,
        },
        undefined,
      ]
    }
  >
    <View
      onMoveShouldSetResponder={[Function]}
      onMoveShouldSetResponderCapture={[Function]}
      onResponderEnd={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderReject={[Function]}
      onResponderRelease={[Function]}
      onResponderStart={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      onStartShouldSetResponderCapture={[Function]}
      style={
        Array [
          Object {
            "flex": 1,
            "flexDirection": "column-reverse",
          },
          Object {
            "backgroundColor": "#000",
          },
        ]
      }
    >
      <View
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <View
          collapsable={undefined}
          pointerEvents="auto"
          style={
            Object {
              "backgroundColor": "#E9E9EF",
              "bottom": 0,
              "left": 0,
              "opacity": 1,
              "position": "absolute",
              "right": 0,
              "shadowColor": "black",
              "shadowOffset": Object {
                "height": 0,
                "width": 0,
              },
              "shadowOpacity": 0.2,
              "shadowRadius": 5,
              "top": 0,
              "transform": Array [
                Object {
                  "translateX": 0,
                },
                Object {
                  "translateY": 0,
                },
              ],
            }
          }
        >
          <View
            loaded={
              Array [
                0,
              ]
            }
            onLayout={[Function]}
            style={
              Array [
                Object {
                  "flex": 1,
                  "overflow": "hidden",
                },
                Object {
                  "flex": 1,
                },
              ]
            }
          >
            <View
              collapsable={undefined}
              onMoveShouldSetResponder={[Function]}
              onMoveShouldSetResponderCapture={[Function]}
              onResponderEnd={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderReject={[Function]}
              onResponderRelease={[Function]}
              onResponderStart={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              onStartShouldSetResponderCapture={[Function]}
              style={
                Object {
                  "alignItems": "stretch",
                  "flex": 1,
                  "flexDirection": "row",
                }
              }
            >
              <View
                style={
                  Object {
                    "bottom": 0,
                    "left": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  }
                }
                testID={undefined}
              >
                <View
                  style={
                    Object {
                      "flex": 1,
                      "overflow": "hidden",
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "backgroundColor": "#fff",
                        "flex": 1,
                      }
                    }
                  >
                    <RCTScrollView
                      contentContainerStyle={
                        Object {
                          "paddingTop": 30,
                        }
                      }
                      style={
                        Object {
                          "backgroundColor": "#fff",
                          "flex": 1,
                        }
                      }
                    >
                      <View>
                        <View
                          style={
                            Object {
                              "alignItems": "center",
                              "marginBottom": 20,
                              "marginTop": 10,
                            }
                          }
                        >
                          <Image
                            source={1}
                            style={
                              Object {
                                "height": 80,
                                "marginLeft": -10,
                                "marginTop": 3,
                                "resizeMode": "contain",
                                "width": 100,
                              }
                            }
                          />
                        </View>
                        <View
                          style={
                            Object {
                              "alignItems": "center",
                              "marginHorizontal": 50,
                            }
                          }
                        >
                          <Text
                            accessible={true}
                            allowFontScaling={true}
                            disabled={false}
                            ellipsizeMode="tail"
                            style={
                              Object {
                                "color": "rgba(0,0,0,0.4)",
                                "fontSize": 14,
                                "lineHeight": 19,
                                "marginBottom": 20,
                                "textAlign": "center",
                              }
                            }
                          >
                            Development mode is enabled, your app will be slower but you can use useful development tools. 
                            <Text
                              accessible={true}
                              allowFontScaling={true}
                              disabled={false}
                              ellipsizeMode="tail"
                              onPress={[Function]}
                              style={
                                Object {
                                  "color": "#2e78b7",
                                  "fontSize": 14,
                                }
                              }
                            >
                              Learn more
                            </Text>
                          </Text>
                          <Text
                            accessible={true}
                            allowFontScaling={true}
                            disabled={false}
                            ellipsizeMode="tail"
                            style={
                              Object {
                                "color": "rgba(96,100,109, 1)",
                                "fontSize": 17,
                                "lineHeight": 24,
                                "textAlign": "center",
                              }
                            }
                          >
                            Get started by opening
                          </Text>
                          <View
                            style={
                              Array [
                                Object {
                                  "backgroundColor": "rgba(0,0,0,0.05)",
                                  "borderRadius": 3,
                                  "paddingHorizontal": 4,
                                },
                                Object {
                                  "marginVertical": 7,
                                },
                              ]
                            }
                          >
                            <Text
                              accessible={true}
                              allowFontScaling={true}
                              disabled={false}
                              ellipsizeMode="tail"
                              style={
                                Array [
                                  Object {
                                    "color": "rgba(96,100,109, 0.8)",
                                  },
                                  Object {
                                    "fontFamily": "space-mono",
                                  },
                                ]
                              }
                            >
                              screens/HomeScreen.js
                            </Text>
                          </View>
                          <Text
                            accessible={true}
                            allowFontScaling={true}
                            disabled={false}
                            ellipsizeMode="tail"
                            style={
                              Object {
                                "color": "rgba(96,100,109, 1)",
                                "fontSize": 17,
                                "lineHeight": 24,
                                "textAlign": "center",
                              }
                            }
                          >
                            Change this text and your app will automatically reload.
                          </Text>
                        </View>
                        <View
                          style={
                            Object {
                              "alignItems": "center",
                              "marginTop": 15,
                            }
                          }
                        >
                          <View
                            accessibilityComponentType={undefined}
                            accessibilityLabel={undefined}
                            accessibilityTraits={undefined}
                            accessible={true}
                            collapsable={undefined}
                            hitSlop={undefined}
                            isTVSelectable={true}
                            nativeID={undefined}
                            onLayout={undefined}
                            onResponderGrant={[Function]}
                            onResponderMove={[Function]}
                            onResponderRelease={[Function]}
                            onResponderTerminate={[Function]}
                            onResponderTerminationRequest={[Function]}
                            onStartShouldSetResponder={[Function]}
                            style={
                              Object {
                                "opacity": 1,
                                "paddingVertical": 15,
                              }
                            }
                            testID={undefined}
                            tvParallaxProperties={undefined}
                          >
                            <Text
                              accessible={true}
                              allowFontScaling={true}
                              disabled={false}
                              ellipsizeMode="tail"
                              style={
                                Object {
                                  "color": "#2e78b7",
                                  "fontSize": 14,
                                }
                              }
                            >
                              Help, it didn’t automatically reload!
                            </Text>
                          </View>
                        </View>
                      </View>
                    </RCTScrollView>
                    <View
                      style={
                        Object {
                          "alignItems": "center",
                          "backgroundColor": "#fbfbfb",
                          "bottom": 0,
                          "left": 0,
                          "paddingVertical": 20,
                          "position": "absolute",
                          "right": 0,
                          "shadowColor": "black",
                          "shadowOffset": Object {
                            "height": -3,
                          },
                          "shadowOpacity": 0.1,
                          "shadowRadius": 3,
                        }
                      }
                    >
                      <Text
                        accessible={true}
                        allowFontScaling={true}
                        disabled={false}
                        ellipsizeMode="tail"
                        style={
                          Object {
                            "color": "rgba(96,100,109, 1)",
                            "fontSize": 17,
                            "textAlign": "center",
                          }
                        }
                      >
                        This is a tab bar. You can edit it in:
                      </Text>
                      <View
                        style={
                          Array [
                            Object {
                              "backgroundColor": "rgba(0,0,0,0.05)",
                              "borderRadius": 3,
                              "paddingHorizontal": 4,
                            },
                            Object {
                              "marginTop": 5,
                            },
                          ]
                        }
                      >
                        <Text
                          accessible={true}
                          allowFontScaling={true}
                          disabled={false}
                          ellipsizeMode="tail"
                          style={
                            Array [
                              Object {
                                "color": "rgba(96,100,109, 0.8)",
                              },
                              Object {
                                "fontFamily": "space-mono",
                              },
                            ]
                          }
                        >
                          navigation/MainTabNavigator.js
                        </Text>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
              <View
                style={null}
                testID={undefined}
              />
              <View
                style={null}
                testID={undefined}
              />
            </View>
            <View
              collapsable={undefined}
              style={
                Object {
                  "backgroundColor": "#F7F7F7",
                  "borderTopColor": "rgba(0, 0, 0, .3)",
                  "borderTopWidth": 0.5,
                  "flexDirection": "row",
                  "height": 49,
                }
              }
            >
              <View
                accessibilityComponentType={undefined}
                accessibilityLabel={undefined}
                accessibilityTraits={undefined}
                accessible={true}
                collapsable={undefined}
                hitSlop={undefined}
                nativeID={undefined}
                onLayout={undefined}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "alignItems": "stretch",
                    "backgroundColor": "rgba(0, 0, 0, 0)",
                    "flex": 1,
                    "justifyContent": "flex-end",
                  }
                }
                testID={undefined}
              >
                <View
                  style={
                    Object {
                      "flexGrow": 1,
                    }
                  }
                >
                  <View
                    collapsable={undefined}
                    style={
                      Object {
                        "alignItems": "center",
                        "bottom": 0,
                        "justifyContent": "center",
                        "left": 0,
                        "opacity": 1,
                        "position": "absolute",
                        "right": 0,
                        "top": 0,
                      }
                    }
                  >
                    <Text
                      accessible={true}
                      allowFontScaling={false}
                      disabled={false}
                      ellipsizeMode="tail"
                      style={
                        Array [
                          Object {
                            "color": "#2f95dc",
                            "fontSize": 28,
                          },
                          Object {
                            "marginBottom": -3,
                          },
                          Object {
                            "fontFamily": "ExponentFont-mock-ionicons",
                            "fontStyle": "normal",
                            "fontWeight": "normal",
                          },
                        ]
                      }
                    >
                      
                    </Text>
                  </View>
                  <View
                    collapsable={undefined}
                    style={
                      Object {
                        "alignItems": "center",
                        "bottom": 0,
                        "justifyContent": "center",
                        "left": 0,
                        "opacity": 0,
                        "position": "absolute",
                        "right": 0,
                        "top": 0,
                      }
                    }
                  >
                    <Text
                      accessible={true}
                      allowFontScaling={false}
                      disabled={false}
                      ellipsizeMode="tail"
                      style={
                        Array [
                          Object {
                            "color": "#ccc",
                            "fontSize": 28,
                          },
                          Object {
                            "marginBottom": -3,
                          },
                          Object {
                            "fontFamily": "ExponentFont-mock-ionicons",
                            "fontStyle": "normal",
                            "fontWeight": "normal",
                          },
                        ]
                      }
                    >
                      
                    </Text>
                  </View>
                </View>
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  collapsable={undefined}
                  disabled={false}
                  ellipsizeMode="tail"
                  style={
                    Object {
                      "backgroundColor": "transparent",
                      "color": "rgba(52, 120, 246, 1)",
                      "fontSize": 10,
                      "marginBottom": 1.5,
                      "textAlign": "center",
                    }
                  }
                >
                  Home
                </Text>
              </View>
              <View
                accessibilityComponentType={undefined}
                accessibilityLabel={undefined}
                accessibilityTraits={undefined}
                accessible={true}
                collapsable={undefined}
                hitSlop={undefined}
                nativeID={undefined}
                onLayout={undefined}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "alignItems": "stretch",
                    "backgroundColor": "rgba(0, 0, 0, 0)",
                    "flex": 1,
                    "justifyContent": "flex-end",
                  }
                }
                testID={undefined}
              >
                <View
                  style={
                    Object {
                      "flexGrow": 1,
                    }
                  }
                >
                  <View
                    collapsable={undefined}
                    style={
                      Object {
                        "alignItems": "center",
                        "bottom": 0,
                        "justifyContent": "center",
                        "left": 0,
                        "opacity": 0,
                        "position": "absolute",
                        "right": 0,
                        "top": 0,
                      }
                    }
                  >
                    <Text
                      accessible={true}
                      allowFontScaling={false}
                      disabled={false}
                      ellipsizeMode="tail"
                      style={
                        Array [
                          Object {
                            "color": "#2f95dc",
                            "fontSize": 28,
                          },
                          Object {
                            "marginBottom": -3,
                          },
                          Object {
                            "fontFamily": "ExponentFont-mock-ionicons",
                            "fontStyle": "normal",
                            "fontWeight": "normal",
                          },
                        ]
                      }
                    >
                      
                    </Text>
                  </View>
                  <View
                    collapsable={undefined}
                    style={
                      Object {
                        "alignItems": "center",
                        "bottom": 0,
                        "justifyContent": "center",
                        "left": 0,
                        "opacity": 1,
                        "position": "absolute",
                        "right": 0,
                        "top": 0,
                      }
                    }
                  >
                    <Text
                      accessible={true}
                      allowFontScaling={false}
                      disabled={false}
                      ellipsizeMode="tail"
                      style={
                        Array [
                          Object {
                            "color": "#ccc",
                            "fontSize": 28,
                          },
                          Object {
                            "marginBottom": -3,
                          },
                          Object {
                            "fontFamily": "ExponentFont-mock-ionicons",
                            "fontStyle": "normal",
                            "fontWeight": "normal",
                          },
                        ]
                      }
                    >
                      
                    </Text>
                  </View>
                </View>
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  collapsable={undefined}
                  disabled={false}
                  ellipsizeMode="tail"
                  style={
                    Object {
                      "backgroundColor": "transparent",
                      "color": "rgba(146, 146, 146, 1)",
                      "fontSize": 10,
                      "marginBottom": 1.5,
                      "textAlign": "center",
                    }
                  }
                >
                  Links
                </Text>
              </View>
              <View
                accessibilityComponentType={undefined}
                accessibilityLabel={undefined}
                accessibilityTraits={undefined}
                accessible={true}
                collapsable={undefined}
                hitSlop={undefined}
                nativeID={undefined}
                onLayout={undefined}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "alignItems": "stretch",
                    "backgroundColor": "rgba(0, 0, 0, 0)",
                    "flex": 1,
                    "justifyContent": "flex-end",
                  }
                }
                testID={undefined}
              >
                <View
                  style={
                    Object {
                      "flexGrow": 1,
                    }
                  }
                >
                  <View
                    collapsable={undefined}
                    style={
                      Object {
                        "alignItems": "center",
                        "bottom": 0,
                        "justifyContent": "center",
                        "left": 0,
                        "opacity": 0,
                        "position": "absolute",
                        "right": 0,
                        "top": 0,
                      }
                    }
                  >
                    <Text
                      accessible={true}
                      allowFontScaling={false}
                      disabled={false}
                      ellipsizeMode="tail"
                      style={
                        Array [
                          Object {
                            "color": "#2f95dc",
                            "fontSize": 28,
                          },
                          Object {
                            "marginBottom": -3,
                          },
                          Object {
                            "fontFamily": "ExponentFont-mock-ionicons",
                            "fontStyle": "normal",
                            "fontWeight": "normal",
                          },
                        ]
                      }
                    >
                      
                    </Text>
                  </View>
                  <View
                    collapsable={undefined}
                    style={
                      Object {
                        "alignItems": "center",
                        "bottom": 0,
                        "justifyContent": "center",
                        "left": 0,
                        "opacity": 1,
                        "position": "absolute",
                        "right": 0,
                        "top": 0,
                      }
                    }
                  >
                    <Text
                      accessible={true}
                      allowFontScaling={false}
                      disabled={false}
                      ellipsizeMode="tail"
                      style={
                        Array [
                          Object {
                            "color": "#ccc",
                            "fontSize": 28,
                          },
                          Object {
                            "marginBottom": -3,
                          },
                          Object {
                            "fontFamily": "ExponentFont-mock-ionicons",
                            "fontStyle": "normal",
                            "fontWeight": "normal",
                          },
                        ]
                      }
                    >
                      
                    </Text>
                  </View>
                </View>
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  collapsable={undefined}
                  disabled={false}
                  ellipsizeMode="tail"
                  style={
                    Object {
                      "backgroundColor": "transparent",
                      "color": "rgba(146, 146, 146, 1)",
                      "fontSize": 10,
                      "marginBottom": 1.5,
                      "textAlign": "center",
                    }
                  }
                >
                  app.json
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
